src/book/03-guides/04-general/04-documentation.mdx

```diff
@@ -0,0 +1,78 @@
+- `@detail` A complete and detailed description of the code.
+- `@param` Details a paramter that is passed to a function.
+- `@return` Details what a function returns.
+- `@retval` Details specific return value meanings, usually for enums, booleans or error codes.
+- `@throws` Details on what a function throws.
+- `@tparam` Details on the template parameter, usually a [named requirement](https://en.cppreference.com/w/cpp/named_req) or other requirements that must be fulfilled.
+
+This is a list of extra things that may be used, see [doxygen](https://www.doxygen.nl/manual/commands.html#cmd_intro) documentation for the full list.
+
+- `@example <file-name>` An example of how the function or class should be used is at `<file-name>`.
+- `@see` Tells the reader to read documentation for another class, function or variable.
+- `@attention` Tells the reader that they must pay attention.
+- `@warning` Tells the reader something that may go wrong.
+- `@note` Tells the reader to take note.
+- `@remark` A remark about the thing.
+- `@todo` Adds a todo to the documentation.
+- `@post` Details a post condition.
+- `@pre` Details a pre condition.
+- `@code` and `@endcode` starts and finishes a code block, like \` and in NUbook.
+- `@copydoc` Copies the documentation from another thing.
+- `@dot` and `@enddot` Creates a dot graph.
+- `@msc` and `@endmsc` Creates a [mscgen](https://www.mcternan.me.uk/mscgen/) diagram.
+- `@startuml` and `@enduml` Creates a [uml](https://plantuml.com/) diagram.
+- `@f$`, `@f[`, `@f]`, `@f{<environment>}{` and `@f}` Creates latex mathematics.
+- `@verbatim` and `@endverbatim` Starts and finishes verbatim output.
+- `@author` The author of this code, this is not used often in NUbots as we generally rely on the git blame instead.
+
+Basic html can be used, like `<b>` and `<em>`.
+
+### Example
+
+This example is more than normally required, but tries to give an example of how most directives work.
+
+```c++
+/**
+ * @brief Is super cool.
+ * @detail Is cooler than all the other classes. It graduated from the university of cool.
+ * @tparam T The first template argument
+ * @tparam U The second template arguemnt
+ */
+template <typename T, typename U>
+class Cool_Class {
+    /// @brief Holds the meaning of life the universe and everything
+    private int a = 42;
+
+    /**
+     * @brief Calculates if this class is cool
+     * @detail Checks against the university of cool records to see if this class graduated.
+     * @param year The year that class graduated.
+     * @throws NotFoundError The class was not found in the university of cool records.
+     * @return Whether or not the class is cool.
+     * @retval true The class is cool.
+     * @retval flase The class is not cool.
+     */
+    bool is_cool(const int& year) const;
+};
+```
+
+## Future
+
+In the future we are planning to use a combination of doxygen and our documentation generator for automatic low level documentation for NUbook.
```

Comment:
we really should rename the "documentation generator". It generates documentation sure, but that name is what I would expect the stuff that generates the html for nubook from the doxygen tags to be called